src/geo/leaflet/leaflet-cartodb-vector-layer-group-view.js

```diff
@@ -27,6 +27,19 @@ var LeafletCartoDBVectorLayerGroupView = CartoDBd3Layer.extend({
       this.setCartoCSS(index, style);
     }, this);
 
+    layerModel.layers.bind('change:meta', function (child, meta) {
+      if (this.model.get('type') === 'namedmap') {
+        var index = child.collection.indexOf(child);
+        if (child.collection.at(0).get('type') === 'tiled') {
+          index++;
+        }
+        this.options.styles[index] = meta.cartocss;
+        if (this.options.styles.indexOf(undefined) === -1) {
```

Comment:
Will this work if a second instance of the map is created (all styles in this.options.styles will have been set after the previous map instance has been created)?